File: frontend/types/project.ts
export interface ProjectStats {
  total_sources: number;
  total_posts: number;
  total_comments: number;
}

export interface Project {
  id: string;
  user_id: string | null;
  name: string;
  description: string | null;
  created_at: string;
  updated_at: string;
}

export interface ProjectDetail extends Project {
  stats: ProjectStats;
}

File: backend/app/schemas/project.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field

from app.schemas.common import ORMModel


class ProjectCreate(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    description: str | None = None


class ProjectResponse(ORMModel):
    id: UUID
    user_id: UUID | None
    name: str
    description: str | None
    created_at: datetime
    updated_at: datetime


class ProjectStats(BaseModel):
    total_sources: int = 0
    total_posts: int = 0
    total_comments: int = 0


class ProjectDetail(ProjectResponse):
    stats: ProjectStats
